app/connectors/FinancialDetailsConnector.scala

```diff
@@ -91,15 +92,17 @@ class FinancialDetailsConnector @Inject()(val http: HttpClient,
 
   // TODO: MFA Credits
   def getFinancialDetails(taxYear: Int, nino: String)
-                         (implicit headerCarrier: HeaderCarrier): Future[FinancialDetailsResponseModel] = {
+                         (implicit headerCarrier: HeaderCarrier, mtdItUser: MtdItUser[_]): Future[FinancialDetailsResponseModel] = {
 
     val dateFrom: String = (taxYear - 1).toString + "-04-06"
     val dateTo: String = taxYear.toString + "-04-05"
 
     val url = getChargesUrl(nino, dateFrom, dateTo)
     Logger("application").debug(s"GET $url")
 
-    http.GET[HttpResponse](url)(httpReads, headerCarrier, implicitly) map { response =>
+    val hc = checkAndAddTestHeader(mtdItUser.path, headerCarrier, appConfig.poaAdjustmentOverrides(), "afterPoaAmountAdjusted")
```

Comment:
* we might need to raise a tech debt to move all this kind of transformation to a different level, like => HeaderExtractor etc;

Reply:
I think we only want this to be called in these 2 specific scenarios (at least, at the moment), so I'm not sure if moving it to the HeaderExtractor is the best place

Reply:
I am going to create a tech debt story for you as I think connectors should not include this type of logic